src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,40 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines = null;
+        if (type.equals(Bulldozer.class)) {
+            machines = new BulldozerProducer().get();
+            return new ArrayList<>(machines);
+        } else if (type.equals(Excavator.class)) {
+            machines = new ExcavatorProducer().get();
+            return new ArrayList<>(machines);
+        } else if (type.equals(Truck.class)) {
+            machines = new TruckProducer().get();
+            return new ArrayList<>(machines);
+        }
+        return new ArrayList<>();
+    }
```

Comment:
```suggestion
        if (type.equals(Bulldozer.class)) {
            return new BulldozerProducer().get();
        } else if (type.equals(Excavator.class)) {
            return new ExcavatorProducer().get();
        } else if (type.equals(Truck.class)) {
            return new TruckProducer().get();
        }
        return new ArrayList<>();
    }
```

Reply:
Compilation error because I need to return a List<Machine> and I return a List<Excavator, Truck or Bulldozer>. I need to leave everything as it is to avoid compilation errors.

Reply:
you're correct, I missed method parametrization